src/convertToObject.js

```diff
@@ -6,7 +6,36 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const splitThis = sourceString.split(';');
+  const result = {};
+
+  for (const oneString of splitThis) {
+    let word = '';
+    let wordTwo = '';
+    let fazeTwo = false;
+
+    for (const symbol of oneString) {
+      if (symbol.toLowerCase() !== symbol.toUpperCase() || symbol === '-') {
+        if (fazeTwo !== true) {
+          word += symbol;
+        }
+      }
+
+      if (fazeTwo === true) {
+        wordTwo += symbol;
+      }
+
+      if (symbol === ':') {
+        fazeTwo = true;
+      }
+    }
+
+    if (word.length !== 0) {
+      result[word] = wordTwo.trim();
+    }
+  }
```

Comment:
The nested `for` loop here is used to parse the individual style declarations. This can be replaced with string manipulation methods such as `split` and `trim` to simplify the code and remove the nested loop.